src/Behat/Testwork/Argument/MixedArgumentOrganiser.php

```diff
@@ -227,7 +227,7 @@ private function filterApplicableTypehintedParameters(array $parameters)
             }
 
             try {
-                $reflectionClass = $parameter->getClass();
+                $reflectionClass = $parameter->getType() ? $parameter->getType()->getName() : null;
```

Comment:
As far as I can see none of these methods throws an exception, does it still need a `try / catch` here?

Also, it looks like this can now also return non-class types, like `array`, `int`, etc. Does that matter?

Reply:
I'm not sure what throws an exception here TBH, it's poorly documented

Reply:
Good point about scalars, what a pain

Reply:
`$parameter->getClass()` is still used on line 286. The needed refactoring might be bigger, to avoid returning `ReflectionParameter` objects here...

Reply:
Yeah I've done a proper search and there are a bunch more places it's used in other files - this was the only one breaking the tests when I ran initially

Reply:
also, be careful about the special `self` and `parent` types. `$parameter->getClass()` had them resolved to the actual class, but `ReflectionNamedType` does not resolve them.